textile/chat-features.textile

```diff
@@ -411,36 +411,56 @@ Presence allows chat room users to indicate to others that they're online, as we
+** @(CHA-T5b)@ This specification point has been removed.
+** @(CHA-T5c)@ @[Testable]@ If the channel state is not @ATTACHED@ or @ATTACHING@, the operation shall throw an @ErrorInfo@ with code @40000@.
+** @(CHA-T5d)@ @[Testable]@ The client shall publish an ephemeral message to the channel with the name field set to @typing.stopped@, the format of which is detailed "here":#realtime-typing-heartbeat.
+** @(CHA-T5e)@ @[Testable]@ On successfully publishing the message in @(CHA-T5d)@, the @CHA-T10@ timer shall be unset.
 * @(CHA-T6)@ Users may subscribe to typing events - updates to a set of clientIDs that are typing. This operation, like all subscription operations, has no side-effects in relation to room lifecycle.
-** @(CHA-T6a)@ @[Testable]@ Users may provide a listener to subscribe to "typing events":#chat-structs-typing-event in a room.
+** @(CHA-T6a)@ @[Testable]@ Users may provide a listener to subscribe to "typing event V2":#chat-structs-typing-event-V2 in a chat room.
 ** @(CHA-T6b)@ @[Testable]@ A subscription to typing may be removed, after which it shall receive no further events.
-** @(CHA-T6c)@ @[Testable]@ When a presence event is received from the realtime client, the Chat client will perform a @presence.get()@ operation to get the current presence set. This guarantees that we get a fully synced presence set. This is then used to emit the typing clients to the subscriber.
-*** @(CHA-T6c1)@ @[Testable]@ If the @presence.get()@ operation fails, then it shall be retried using a backoff with jitter, up to a timeout of 30 seconds.
-*** @(CHA-T6c2)@ @[Testable]@ If multiple presence events are received resulting in concurrent @presence.get()@ calls, then we guarantee that only the "latest" event is emitted. That is to say, if presence event A and B occur in that order, then only the typing event generated by B's call to @presence.get()@ will be emitted to typing subscribers.
-* @(CHA-T7)@ @[Testable]@ Users may subscribe to discontinuity events to know when there's been a break in typing indicators. Their listener will be called when a discontinuity event is triggered from the room lifecycle. For typing, there shouldn't need to be user action as the underlying core SDK will heal the presence set.
+** @(CHA-T6c)@ This specification point has been removed, it has been superseded by @CHA-T6d@.
+*** @(CHA-T6c1)@ This specification point has been removed, it has been superseded by @CHA-T6d@.
+*** @(CHA-T6c2)@ This specification point has been removed, it has been superseded by @CHA-T6d@.
+* @(CHA-T13)@ When a typing event (@typing.start@ or @typing.stop@) is received from the realtime client, the Chat client shall emit appropriate events to the user.
+** @(CHA-T13a)@ @[Testable]@ If the typing event received from the server is malformed, i.e does not conform to the expected "payload":#realtime-typing-events (unknown fields should be ignored), then it shall not be handled and emitted to the subscriber.
```

Comment:
Can we make this explicit, 
```suggestion
** @(CHA-T13a)@ @[Testable]@ If the typing event received from the server has null or empty `msg.clientId`, then it shall not be handled and emitted to the subscriber.
```
This is the only valid we implemented as a part of kotlin typing. Seems other checks are not needed =>
https://github.com/ably/ably-chat-kotlin/blob/9b4eb93c3b9e1b008c5a511da896f11172ff8281/chat-android/src/main/java/com/ably/chat/Typing.kt#L196.
Check for `msg.name` is not needed, since we already register for `typing.started` and `typing.stopped`, so `msg.name` will always be either of those.
`encoding` will be dependent on underlying pub-sub SDK, `timestamp` and `extras` are not needed to be check wdyt

Reply:
Updated typing message format -> https://github.com/ably/specification/pull/277/files#r2019755781.